packages/react-router-dom/index.tsx

```diff
@@ -657,12 +657,13 @@ export const NavLink = React.forwardRef<HTMLAnchorElement, NavLinkProps>(
         : null;
       toPathname = toPathname.toLowerCase();
     }
-
+    const endSlashPosition = (toPathname.length > 1 && toPathname[toPathname.length - 1] === "/") ? toPathname.length - 1 : toPathname.length;
```

Comment:
We can use `endsWith` here for a little more clarity:

```suggestion
    const endSlashPosition =
      toPathname.length > 1 && toPathname.endsWith("/")
        ? toPathname.length - 1
        : toPathname.length;
```